### ISSUE TITLE
require-throws triggered by arrow function without body

### ISSUE THREAD

**Issue by homer0** [gajus/eslint-plugin-jsdoc#597]
## Expected behavior

There shouldn't be any errors on a fully documented function (arrow expression) without a body.

## Actual behavior

The plugin is telling me that I'm missing a `@throws` declaration when there's no `throw`.

## ESLint Config

```js
'jsdoc/require-throws': 'error',
```

(I have a lot of other rules for the Airbnb preset, but I'm almost sure it's not affecting this issue)

## ESLint sample

```js
/**
 * @typedef {Function} Add
 * @param {number} n Something.
 * @returns {number}
 */

/**
 * Something.
 *
 * @param {number} base  Something.
 * @returns {Add}
 */
const sum = (base) => (n) => base + n;
```

## Environment

* Node version: 10.20
* ESLint version 7.3.1
* `eslint-plugin-jsdoc` version: 28.6.0

---

I believe the issue is on `hasThrowValue`, when the node is an `ArrowFunctionExpression`, it checks for `expression`... that's always `true`, and unrelated to a `throw`, if the function doesn't have a body.

Writing `sum` this way doesn't trigger the rule:

```js
const sum = (base) => {
  return (n) => base + n;
};
```

But it triggers [arrow-body-style](https://eslint.org/docs/rules/arrow-body-style) :P.

**Comment by brettz9** [comment]
Was apparently a copy-paste bug (from the `hasReturnValue` function where `expression` on a function successfully denotes an (arrow) return value, but does not denote a throw statement).

Now fixed.

I also tweaked the rule to avoid something like the following reporting errors:

```js
const nested = () => () => {throw new Error('oops');};
```

Since `nested` does not directly throw, its docs should not be indicating that it throws.

Note that we do not currently catch:

```js
      const nested = () => {
        function a () {
          throw new Error('oops');
        }
        a();
      };
```

...nor does our current comment-detection approach (`getJSDocComment` in source) check comment blocks directly above expressions, so one couldn't get the nested block to err (though it probably should be made as an option):

```js
const nested = () => 
  /**
   *
   */
  () => {throw new Error('oops');};
```

**Comment by homer0** [comment]
@brettz9 thank you so much :D!! Everything works now, and the change you made for the nested scenario makes a lot of sense too.
